ledger/eval.go

```diff
@@ -675,7 +675,7 @@ func (eval *BlockEvaluator) TransactionGroup(txads []transactions.SignedTxnWithA
 
 // prepareEvalParams creates a logic.EvalParams for each ApplicationCall
 // transaction in the group
-func (eval *BlockEvaluator) prepareEvalParams(txgroup []transactions.SignedTxnWithAD) (res []*logic.EvalParams) {
+func (eval *BlockEvaluator) prepareEvalParams(txgroup []transactions.SignedTxnWithAD, spec *transactions.SpecialAddresses) (res []*logic.EvalParams) {
```

Comment:
why do you need to pass it? `BlockEvaluator` has `eval.block.BlockHeader` data. For unit testing?

Reply:
I was trying to avoid allocating the little `spec` object twice.  The caller needs to make it, then pass it in here and to eval.transaction. So I make it first and use it for both.

Reply:
This is now in eval.specials so no need to allocate or pass.